### PR TITLE
Upgrade dartdoc to 0.12.0

### PR DESCRIPTION
**Pull request by jcollins-g**
Including @devoncarew's change from #10142, change version to 0.12.0.

See dartdoc changelog here:  https://github.com/dart-lang/dartdoc/releases/tag/v0.12.0

Generated by 0.11.1:  http://jcollins1.pdx.corp.google.com:8011/
Generated by 0.12.0 + CSS changes (this PR):  http://jcollins1.pdx.corp.google.com:8012/

### CODE REVIEW

File: dev/docs/styles.html [hunk 1]
```diff
@@ -14,7 +15,17 @@
...
+    background-color: white;
+    color: #424242;
+  }
+
+  nav.navbar {
+    min-height: 57px;
+    padding: 6px 0;
+  }
+
+  nav.nav.navbar-fixed-top {
```
Comment (devoncarew) [inline dev/docs/styles.html]: I think the selector here needs to be `header.header-fixed nav.navbar-fixed-top` - the shadow under the header doesn't quite match the new one on flutter.io.

Comment (jcollins-g) [inline dev/docs/styles.html]: Done, and updated http://jcollins1.pdx.corp.google.com:8012/.  It does look much more like the one on flutter.io now.

**Comment by devoncarew** [comment]
lgtm, but you may also want to get a stamp from @Hixie

**Comment by Hixie** [comment]
LGTM

**Pull request merged**